release/artifacts.yaml

```diff
@@ -1,25 +1,24 @@
-- target: //daml-lf/archive:daml_lf_archive_java
-  type: jar
-  javadoc-jar: daml_lf_archive_java_javadoc.jar
-  src-jar: daml_lf_archive_java_srcjar.jar
-  mavenUpload: True
 - target: //daml-lf/data:data
   type: jar-scala
-- target: //compiler/damlc/jar:damlc_jar
-  type: jar-deploy
-  platformDependent: True
-- target: //daml-lf/archive:daml_lf_archive_reader
-  type: jar-scala
-- target: //daml-lf/archive:daml_lf_archive_protos_zip
+- target: //daml-lf/archive:daml_lf_archive_java_proto
+  type: jar-lib
+  javadoc-jar: daml_lf_archive_java_proto_javadoc.jar
```

Comment:
Do we no longer need the `src-jar` attribute here?

Reply:
I changed how the package is generated. 
Like that the haskell releaser seems to find the sources without hint.  

```
  remy@remy-ThinkPad:~/work/daml$ ./ci/release.sh 
  ...
  Make dry run of release
  step: release artifacts got stored in /tmp/tmp.t7KbRa77vq
  remy@remy-ThinkPad:~/work/daml$ ls /tmp/tmp.t7KbRa77vq/com/digitalasset/daml-lf-archive-java-proto/100.13.29/
  daml-lf-archive-java-proto-100.13.29.jar  
  daml-lf-archive-java-proto-100.13.29-javadoc.jar 
  daml-lf-archive-java-proto-100.13.29.pom
  daml-lf-archive-java-proto-100.13.29-sources.jar
  remy@remy-ThinkPad:~/work/daml$ 
```